Server/ConcordiaCurriculumManager/Security/Requirements/Handlers/AdminHandler.cs

```diff
@@ -0,0 +1,64 @@
+﻿using ConcordiaCurriculumManager.Models.Users;
+using ConcordiaCurriculumManager.Services;
+using Microsoft.AspNetCore.Authorization;
+
+namespace ConcordiaCurriculumManager.Security.Requirements.Handlers;
+
+public class AdminHandler : IAuthorizationHandler
+{
+    private readonly ILogger<AdminHandler> _logger;
+    private readonly IUserService _userService;
+
+    public AdminHandler(ILogger<AdminHandler> logger, IUserService userService)
+    {
+        _logger = logger;
+        _userService = userService;
+    }
+
+    public async Task HandleAsync(AuthorizationHandlerContext context)
+    {
+        var pendingRequirements = context.PendingRequirements.ToList();
+        var result = pendingRequirements.Select(requirement => CheckIfAdmin(context, requirement));
+        await Task.WhenAll(result);
+    }
+
+    private async Task CheckIfAdmin(AuthorizationHandlerContext context, IAuthorizationRequirement requirement)
+    {
+        if (requirement is not GroupMasterOrAdminRequirement)
+        {
+            return;
+        }
+
+        _logger.LogInformation("Evaluting AdminHandler");
+
+        if (context.User.Identity is null || !context.User.Identity.IsAuthenticated)
```

Comment:
Can you not just call `_userAuthenticationService.GetCurrentUserClaim(Claims.Roles);` and skip all this logic, including a DB access? All the roles are already encoded in the JWT.

Reply:
I can. However, normally role checking is not done using the claims on the backend if the JWT is not updated in case of a role change. I'm not sure if we plan on allowing other admins to demote admins to regular users so I just added this logic.